internal/testing/releasehelper.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Helper tool for creating new releases of the Go CDK.
+// Run from the repository root:
+// $ go run internal/testing/releasehelper.go <flags>
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// TODO(eliben): add bumpversion flag that bumps version by 1, or to a version
+// given on the command line
+var doReplace = flag.Bool("replace", false, "add replace directives to root")
+var doDropReplace = flag.Bool("dropreplace", false, "drop replace directives to root")
+
+// cmdCheck invokes the command given in s, echoing the invocation to stdout.
+// It checks that the command was successful and returns its standard output.
+// If the command returned a non-0 status, log.Fatal is invoked.
+func cmdCheck(s string) []byte {
+	fmt.Printf(" -> %s\n", s)
+	fields := strings.Fields(s)
+	if len(fields) < 1 {
+		log.Fatal("Expected \"command <arguments>\"")
+	}
+	b, err := exec.Command(fields[0], fields[1:]...).Output()
+	if exiterr, ok := err.(*exec.ExitError); ok {
+		log.Fatalf("%s; stderr: %s\n", err, string(exiterr.Stderr))
+	} else if err != nil {
+		log.Fatal("exec.Command", err)
+	}
+	return b
+}
+
+// These types are taken from "go mod help edit", in order to parse the JSON
+// output of `go mod edit -json`.
+type GoMod struct {
+	Module  Module
+	Go      string
+	Require []Require
+	Exclude []Module
+	Replace []Replace
+}
+
+type Module struct {
+	Path    string
+	Version string
+}
+
+type Require struct {
+	Path     string
+	Version  string
+	Indirect bool
+}
+
+type Replace struct {
+	Old Module
+	New Module
+}
+
+// parseModuleInfo parses module information from a go.mod file at path.
+func parseModuleInfo(path string) GoMod {
+	rawJson := cmdCheck("go mod edit -json " + path)
+	var modInfo GoMod
+	err := json.Unmarshal(rawJson, &modInfo)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return modInfo
+}
+
+func runOnGomod(path string) {
```

Comment:
Consider passing the module path instead of the full path to the `go.mod` file? It's easy to construct the path to the `go.mod` here instead, and I think it's clearer to do it that way than to call `filepath.Dir(path)` down below.

E.g.,

```
func runOnModule(modulepath string) {
  gomodpath = filepath.Join(modulepath, "go.mod")
  ...
```

Reply:
Done